packages/language-service/ivy/definitions.ts

```diff
@@ -81,27 +69,79 @@ export class DefinitionBuilder {
           });
         }
         if (symbol.kind === SymbolKind.Variable) {
-          definitions.push(...this.getDefinitionInfos(symbol.shimLocation));
+          definitions.push(...getDefinitions(symbol));
         }
         return definitions;
       }
       case SymbolKind.Expression: {
-        const {shimLocation} = symbol;
-        return this.getDefinitionInfos(shimLocation);
+        return getDefinitions(symbol);
       }
     }
   }
 
-  private getDefinitionsForSymbols(symbols: {shimLocation: ShimLocation}[]) {
-    const definitions: ts.DefinitionInfo[] = [];
-    for (const {shimLocation} of symbols) {
-      definitions.push(...this.getDefinitionInfos(shimLocation));
+  getTypeDefinitionAtPosition(fileName: string, position: number):
+      readonly ts.DefinitionInfo[]|undefined {
+    const {symbol, node} = this.getSymbolAndNodeAtPosition(fileName, position);
+    if (symbol === undefined || node === undefined) {
+      return undefined;
+    }
+
+    const getTypeDefinitions = (...symbols: {shimLocation: ShimLocation}[]) =>
```

Comment:
nit (google style guide):
```suggestion
    const getTypeDefinitions = (...symbols: Array<{shimLocation: ShimLocation}>) =>
```